src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -72,24 +77,11 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
             return children;
         } catch (error) {
             const parsedError: IParsedError = parseError(error);
-
             if (this.parent.azureName && parsedError.errorType === invalidCredentialsErrorType) {
                 void context.ui.showWarningMessage(localize('couldNotConnect', 'Could not connect to "{0}": {1}', this.parent.label, parsedError.message), { stepName: 'loadPostgresDatabases' });
-                const credentialsTreeItem: AzExtTreeItem = new GenericTreeItem(this, {
```

Comment:
This feels like too many changes to squeeze in right before a release, after we've already done testing. Flexible server is still in preview so the experience doesn't need to be perfect, especially since this could actually be fixed on the platform side (if they threw the firewall error instead of ETIMEDOUT).

IMO our top priority should be to avoid a regression for single server. While your fix might be okay, front-loading the logic does risk causing perf issues.

Reply:
I agree with you on this, checking for firewall rule with every load doesn't seem efficient. 
I was thinking if we highlight that Flexible Servers are added in Preview in the ChangeLog and stick with the previous fix (#1890), that should be enough for this release.